clusterloader2/api/validation/validation.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+
+	"k8s.io/perf-tests/clusterloader2/api"
+)
+
+// VerifyConfig does simple validatio for config and all included structures.
+// Returns error is any condition is not satisfied.
+func VerifyConfig(c *api.Config) error {
```

Comment:
At the high level, maybe we should make the validation more similar to what we do in k8s?

In particular, we can vendor:
k8s.io/apimachinery/pkg/util/validation/...
package, and then make validation methods more like:
https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/validation/validation.go#L330
(take fldPath as argument,  return ErrorList, use those field.Required, field.Invalid etc error)

Reply:
I didn't know about k8s.io/apimachinery/pkg/util/validation/... - it looks quite useful.
I've change the code, so it uses ErrorList.